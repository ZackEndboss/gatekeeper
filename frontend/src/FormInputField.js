import React from 'react';
import {Field} from 'redux-form';
import {FormGroup, ControlLabel, Col} from 'react-bootstrap';
import PropTypes from 'prop-types';

export default class FormInputField extends React.Component {
    render() {
        let {name, label, placeholder, labelProps, inputColProps, autoFocus} = this.props;
        placeholder = placeholder || label;
        autoFocus = autoFocus || false;

        return (
            <FormGroup controlId={name}>
                <Col componentClass={ControlLabel} {...(labelProps || {})} >
                    {label}
                </Col>
                <Col {...(inputColProps || {})}>
                    <Field id={name}
                           name={name}
                           className='form-control'
                           component='input'
                           type='text'
                           autoFocus={autoFocus}
                           placeholder={placeholder}/>
                </Col>
            </FormGroup>
        );
    }
}

FormInputField.propTypes = {
    name: PropTypes.string.isRequired,
    label: PropTypes.string,
    placeholder: PropTypes.string,
    labelProps: PropTypes.object,
    autoFocus: PropTypes.bool,
    inputColProps: PropTypes.object
};
