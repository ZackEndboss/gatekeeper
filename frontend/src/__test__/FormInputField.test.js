import React from 'react';
import FormInputField  from '../FormInputField';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';
import {MockStore} from './mockStore';
import {Provider} from 'react-redux';
import {Field} from 'redux-form';
import {reduxForm} from 'redux-form';

jest.mock('redux-form/immutable', () => ({
    Field: 'Field',
    reduxForm: (redux_form_configuration) => (wrapped_form_component) => wrapped_form_component
}));

describe('FormInputField component', () => {
    it('should contain all components', () => {
        const Decorated = reduxForm({form: 'testForm'})(
            FormInputField
        );
        const storeMock = MockStore({});
        const component = renderer.create(
            <Provider store={storeMock}>
                <Decorated name='someInputField'
                           label='someLabel'
                           inputProps={{
                               type: 'text',
                               component: 'input'
                           }}
                           placeholder='somePlaceholder'/>
            </Provider>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
