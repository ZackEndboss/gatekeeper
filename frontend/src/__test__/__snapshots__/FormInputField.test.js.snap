// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInputField component should contain all components 1`] = `
<div
  className="form-group"
>
  <label
    className="control-label"
    htmlFor="someInputField"
  >
    someLabel
  </label>
  <div
    className=""
  >
    <input
      autoFocus={false}
      className="form-control"
      id="someInputField"
      name="someInputField"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      placeholder="somePlaceholder"
      type="text"
      value=""
    />
  </div>
</div>
`;
